### ISSUE TITLE
[Question] Can I load a the huggingface llama model aswell?

### ISSUE THREAD

**Issue by maxkraft7** [ggerganov/llama.cpp#708]
I have downloaded the llama-model from [here](https://huggingface.co/decapoda-research/llama-7b-hf). There it got converted to be compatible with pytorch. But the biggest advantage is that it is actually available. The magnet link from that [PR](https://github.com/facebookresearch/llama/pull/73) has no trackers so it's not starting to download at least for me. And the IPFS files always have a different checksum when I download them. So since I only have the huggingface-version is it possible use their model with llama.cpp somehow?

**Comment by KASR** [comment]
With some modifications you might be able to use this: [alpaca-convert-colab](https://github.com/taiyou2000/alpaca-convert-colab/blob/main/alpaca-convert-colab-fixed.ipynb), I haven't tested it however. 

You would only need to run the first two blocks up to `torch.save(model, "consolidated.01.pth")` (and renaming the file) once you have the .pth and the params.json files you can use the conversion scripts of llama cpp to obtain the ggjt files.

**Comment by MillionthOdin16** [comment]
Here's what I used last night. I'm not sure if this is the same thing KASR is mentioning or not.

[Convert.py](https://github.com/ggerganov/llama.cpp/issues/707#issuecomment-1493365088)

**Comment by Technetium1** [comment]
@maxkraft7 I enjoy automatic tracker additions via https://github.com/c0re100/qBittorrent-Enhanced-Edition but you can add these manually to _anything_ https://github.com/ngosang/trackerslist/blob/master/trackers_all.txt

**Comment by ahoho** [comment]
I did it in two steps: I modified (export_state_dict_checkpoint.py)[https://github.com/tloen/alpaca-lora/blob/main/export_state_dict_checkpoint.py] from alpaca-lora to create a consolidated file, then a slightly modified `convert-pth-to-ggml.py` from `llama.cpp`. Don't forget to clean up the intermediate files :)

Here's a gist with my changes. Tested with the 7B-30B LLaMA models from decapoda-research

I mentioned it [elsewhere](https://www.github.com/ggerganov/llama.cpp/issues/397#issuecomment-1493381230), but if you're quantizing, I've had better results with `q4_0` than `q4_1`. YMMV though
